### PR TITLE
Theme starter improvements

### PR DESCRIPTION
**Pull request by filrak**
Added few new things to theme starter
- Very basic microcart
- Fully working product tile
- Fully working category page with filters
- Dumb header

### CODE REVIEW

File: src/themes/theme-starter/pages/Category.vue [hunk 1]
```diff
@@ -2,17 +2,27 @@
...
+    <div class="row">
+      <div class="col-md-3">
+        <Sidebar :filters="filters.available"/>
+      </div>
+      <div class="col-md-9">
+        <div class="row">
+          <div v-if="isCategoryEmpty" class="col-12">
+            Category is empty
+          </div>
+          <product-tile v-for="product in products" :product="product" :key="product.name" class="col-md-3 col-sm-4 col-xs-12"/>
```
Comment (szafran89) [inline src/themes/theme-starter/pages/Category.vue]: please split to multiple lines

File: src/themes/theme-starter/App.vue [hunk 2]
```diff
@@ -1,16 +1,22 @@
 <template>
   <div id="app">
-    <router-view/>
+    <Main-header />
```
Comment (szafran89) [inline src/themes/theme-starter/App.vue]: please change to `<main-header />`

File: src/themes/theme-starter/App.vue [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
+    <Microcart />
```
Comment (szafran89) [inline src/themes/theme-starter/App.vue]: please change to `<microcart />`

File: src/themes/theme-starter/components/Filters/ColorSelector.vue [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <button
+    @click="switchFilter(id, label)"
+    :class="{ active: active }"
+    :aria-label="$t('Select color ') + label"
+  >
+    <div
+      :style="colorFrom(label)"
+    />
```
Comment (szafran89) [inline src/themes/theme-starter/components/Filters/ColorSelector.vue]: IMO we can keep it in one line `<div :style="colorFrom(label)"/>`

File: src/themes/theme-starter/components/Category/Sidebar.vue [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          v-for="(size, index) in filter"
+          :key="index"
+          :id="size.id"
+          :label="size.label"
+        />
+      </div>
+      <div v-else-if="filterIndex==='price'">
+        <price-selector
+          context="category"
+          :attribute_code="price"
```
Comment (szafran89) [inline src/themes/theme-starter/components/Category/Sidebar.vue]: Can you change it to kebab-case?

**Comment by Igloczek** [comment]
@filrak I merged current `develop` and fixed conflicts to this code, but since you coded it on your own `develop` it makes it harder to work on this since names are conflicting, so I created branch `feature/theme-starter` in this repository.

**Comment by filrak** [comment]
@Igloczek thanks! I'll work on this one. I'm closing this PR then

**Comment by filrak** [comment]
Need to wait for https://github.com/DivanteLtd/vue-storefront/pull/1177

**Pull request closed without merging**